engine/src/main/java/org/camunda/bpm/engine/externaltask/ExternalTask.java

```diff
@@ -121,6 +123,12 @@
    */
   long getPriority();
 
+  /**
+   * Returns a map containing all custom extension properties of the external task.
+   * 
+   * @return the properties or {@link Collections#emptyMap()} if no properties were configured
```

Comment:
I guess it's self-explanatory that a map can be empty and it's an implementation detail as well. 
I think the "never null" contract is more interesting for consumers.
```suggestion
   * @return the properties, never <code>null</code>
```